src/misc/msccl/msccl_lifecycle.cc

```diff
@@ -304,7 +304,7 @@ static ncclResult_t mscclInternalSchedulerSelectAlgo(int rank, struct mscclSched
     auto &m = status.algoMetas[i];
     size_t nBytes = param->count * ncclTypeSize(param->dataType) * m.sizeMultiplier;
     bool msgSizeIsValid =
-      param->count > 0 && (param->count % m.nChunksPerLoop) == 0 &&
+      param->count > 0 && ((param->count * m.sizeMultiplier) % m.nChunksPerLoop) == 0 &&
```

Comment:
How is the size multiplier calculated? Why is it > 1 for reduce-scatter?

Reply:
sizeMultiplier = nGPUs for all-gather, reduce-scatter and all-to-all, which is assigned in [msccl_parser.cc](https://github.com/ROCm/rccl/blob/develop/src/misc/msccl/msccl_parser.cc#L330-L373). For those collectives count is per-rank according to API definition in [nccl.h.in](https://github.com/ROCm/rccl/blob/develop/src/nccl.h.in).